tests/arp/test_neighbor_mac.py

```diff
@@ -0,0 +1,181 @@
+import logging
+import pytest
+import time
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.config_reload import config_reload
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('ptf')
+]
+class TestNeighborMac:
+    """
+        Test handling of neighbor MAC in SONiC switch with PTF docker 
+    """
+    PTF_HOST_IF = "eth0"
+    DUT_ETH_IF = "Ethernet0"
+    PTF_HOST_IP = "20.0.0.2"
+    PTF_HOST_NETMASK = "255.255.255.0"
+    DUT_INTF_IP = "20.0.0.1"
+    DUT_INTF_NETMASK = "24"
+    TEST_MAC = ["00:c0:ca:c0:1a:05", "00:c0:ca:c0:1a:06"]
+
+    @pytest.fixture(scope="module", autouse=True)
+    def interfaceConfig(self, duthost):
+        """
+            Configures and Restores DUT configuration after test completes
+
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Returns:
+                None
+        """
+        logger.info("Configure the DUT interface, start interface, add IP address")
+        self.__startInterface(duthost)
+        self.__configureInterfaceIp(duthost, action="add")
+
+        yield
+
+        logger.info("Restore the DUT interface config, remove IP address")
+        self.__configureInterfaceIp(duthost, action="remove")
+
+    @pytest.fixture(params=[0, 1])
+    def macIndex(self, request):
+        """
+            Parameterized fixture for macIndex
+
+            Args:
+                request: pytest request object
+
+            Retruns:
+                macIndex (int): index of the mac address used from TEST_MAC
+        """
+        yield request.param
```

Comment:
Why use here yield in case of any teardown is not perform. Return could be used instead